auth: prompt the user to open the browser

sudomateo opened the pull request:
Previously the `oxide auth login` command would automatically open the
user's browser which can be a jarring user experience.

This patch updates `oxide auth login` to prompt the user to open the
browser instead of just opening the browser automatically. Users passing
`--no-browser` will not receive this prompt and will instead have the
URL printed to standard output.

Closes: #88

sudomateo committed:
auth: prompt the user to open the browser

Previously the `oxide auth login` command would automatically open the
user's browser which can be a jarring user experience.

This patch updates `oxide auth login` to prompt the user to open the
browser instead of just opening the browser automatically. Users passing
`--no-browser` will not receive this prompt and will instead have the
URL printed to standard output.

Closes: #88

ahl commented:
Sorry for the delay on this. Looks good: any thoughts on how we might add a test?

sudomateo commented:
> Sorry for the delay on this. Looks good: any thoughts on how we might add a test?

No worries! I can add a unit test to `wait_for_enter`, but I wasn't entirely sure how to best test the changes to the `auth` command. I was hoping I could get some suggestions on it.

I see there are tests in `cli/tests` that I could probably use as inspiration.

ahl commented:
Give it a shot? if nothing seems to work well then we'll merge as-is. Thanks again!

sudomateo commented:
> Give it a shot? if nothing seems to work well then we'll merge as-is. Thanks again!

Will do! Busy week this week but I'll work on it sometime next week. I'll ping here when updated.

sudomateo commented:
Sorry for the delay on this. I finally got some time to work on this. I refactored the function to be generic and added unit tests. If there's time tomorrow I'll take a look at adding an integration test that actually runs `oxide auth login` with mocked data.

sudomateo committed:
tests: add integration test for `auth` subcommand

sudomateo commented:
I added integration tests. This is ready for another review. There _might_ be a need to update how the `open::that` is called since I'd imagine we'd want to skip opening the browser in integration tests, but we can discuss that during review.

sudomateo commented:
Rebased and updated the branch. `cargo test` is passing locally. I don't have permission to start CI/CD here.

ahl commented:
Not sure where this is. I'm going to close it for now, but we can re-open it if there's interest in recussitating it.
